server/bin/org_migrator/org_migrator.py

```diff
@@ -338,7 +338,51 @@ def do_export(self):
         if self.exported:
             return True
 
-        self._export_query('cp2_products.json', 'cp2_products', 'SELECT p.* FROM cp2_products p JOIN cp2_owner_products op ON op.product_uuid = p.uuid WHERE op.owner_id = %s', (self.org_id,))
+        # Product exporting is recursive as of CP4.0, due to the derived mapping. We can skip it for the
+        # provided product mapping, since that's a separate table which we can populate last, but derived
+        # products force us to pull product info in the order in which it is referenced.
+        cursor = self.db.execQuery('SELECT p.* FROM cp2_products p JOIN cp2_owner_products op ON op.product_uuid = p.uuid WHERE op.owner_id = %s', (self.org_id,))
+        columns = get_cursor_columns(self.db, cursor)
+        column_types = get_cursor_column_types(self.db, cursor)
+
+        uuid_idx = columns.index('uuid')
+        derived_uuid_idx = columns.index('derived_product_uuid')
+
+        products = { row[uuid_idx]: row for row in cursor }
+        cursor.close()
+
+        ordered = []
+        insertion_order = {}
+
+        def insert_row(row):
+            if row[uuid_idx] in insertion_order:
+                # Row already inserted; skip it
+                return
+
+            if row[derived_uuid_idx] is not None:
+                parent_row = products[row[derived_uuid_idx]]
+
+                if parent_row is None:
```

Comment:
Would `child_row` not be a more appropriate name for this variable, since the derived product is child of a main product?

Reply:
yes, fixed